lib/sync/pull-request.js

```diff
@@ -1,57 +1,89 @@
+const url = require('url');
 const transformPullRequest = require('./transforms/pull-request');
-const { getPullRequests: getPullRequestQuery } = require('./queries');
-const { logger } = require('probot/lib/logger');
-
 const statsd = require('../config/statsd');
 
+/**
+ * @typedef {object} RepositoryObject
+ * @property {number} id
+ * @property {string} name
+ * @property {object} owner
+ * @property {string} owner.login
+ * @property {string} html_url
+ * @property {string} full_name
+ */
+
+/**
+ * Find the next page number from the response headers.
+ *
+ * @param {{link: string}} headers - The headers from the API response.
+ * @returns {number|undefined} The next page number.
+ */
+function getNextPage(headers = {}) {
+  const nextUrl = ((headers.link || '').match(/<([^>]+)>;\s*rel="next"/) || [])[1];
```

Comment:
I stole this directly from the paginate plugin https://github.com/octokit/plugin-paginate-rest.js/blob/master/src/iterator.ts#L40-L42